generators/gradle/index.cjs

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2013-2021 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable consistent-return */
+const chalk = require('chalk');
+const { mixBlueprintGenerator } = require('generator-jhipster/support');
+
+const { GENERATOR_PROJECT_NAME, GENERATOR_JAVA, GENERATOR_GRADLE } = require('../generator-list');
+const { files } = require('./files.cjs');
+const { GRADLE, GRADLE_VERSION } = require('./constants.cjs');
+const { BUILD_TOOL } = require('../java/constants.cjs');
+
+const MixedGenerator = mixBlueprintGenerator(GENERATOR_PROJECT_NAME, GENERATOR_JAVA);
+
+module.exports = class extends MixedGenerator {
+  constructor(args, opts, features) {
+    super(args, opts, { jhipsterModular: true, unique: 'namespace', ...features });
+
+    // Register options available to cli.
+    if (!this.fromBlueprint) {
+      this.registerCommonOptions();
+      this.registerProjectNameOptions();
+    }
+
+    if (this.options.help) return;
+
+    this.config.defaults({
+      [BUILD_TOOL]: GRADLE,
+    });
+
+    if (this.options.defaults) {
+      this.configureProjectName();
```

Comment:
Couldn't it be deduced from the `MixedGenerator` constant

Reply:
Or even better: called automatically via the dependency chain (anyway, I don't know if it's feasible)

Reply:
> Couldn't it be deduced from the `MixedGenerator` constant

Need to think about it.

> Or even better: called automatically via the dependency chain (anyway, I don't know if it's feasible)

It's a polymorphism, so some methods can be chained from more than 1 parent, the order cannot be fine tuned.
Not sure if we shouldn't do for this reason.

Reply:
Done.